nbs/utils.ipynb

```diff
@@ -547,6 +556,159 @@
     "        raise ValueError(f'The following values are missing from the index: {missing}')\n",
     "    return idxs"
    ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# 5. Conformal Prediction"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#| export\n",
+    "\n",
+    "class ConformalIntervals:\n",
+    "    \"\"\"Class for storing conformal intervals metadata information.\"\"\"\n",
+    "\n",
+    "    def __init__(\n",
+    "        self,\n",
+    "        n_windows: int = 2,\n",
+    "        method: str = \"conformal_distribution\",\n",
+    "    ):\n",
+    "        \"\"\" \n",
+    "        n_windows : int\n",
+    "            Number of windows to evaluate.\n",
+    "        method : str, default is conformal_distribution\n",
+    "            One of the supported methods for the computation of conformal prediction:\n",
+    "            conformal_error or conformal_distribution\n",
+    "        \"\"\"\n",
+    "        if n_windows < 2:\n",
+    "            raise ValueError(\n",
+    "                \"You need at least two windows to compute conformal intervals\"\n",
+    "            )\n",
+    "        allowed_methods = [\"conformal_error\", \"conformal_distribution\"]\n",
+    "        if method not in allowed_methods:\n",
+    "            raise ValueError(f\"method must be one of {allowed_methods}\")\n",
+    "        self.n_windows = n_windows\n",
+    "        self.method = method\n",
+    "\n",
+    "    def __repr__(self):\n",
+    "        return f\"ConformalIntervals(n_windows={self.n_windows}, method='{self.method}')\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#| export\n",
+    "\n",
+    "def add_conformal_distribution_intervals(\n",
+    "    fcst_df: DataFrame, \n",
+    "    cs_df: DataFrame,\n",
+    "    model_names: List[str],\n",
+    "    level: List[Union[int, float]],\n",
+    "    cs_n_windows: int,\n",
+    "    n_series: int,\n",
+    "    horizon: int,\n",
+    ") -> DataFrame:\n",
+    "    \"\"\"\n",
+    "    Adds conformal intervals to a `fcst_df` based on conformal scores `cs_df`.\n",
+    "    `level` should be already sorted. This strategy creates forecasts paths\n",
```

Comment:
Do we enforce the sorting before calling `add_conformal_distribution_intervals`?